src/nanosvg.h

```diff
@@ -809,17 +816,28 @@ static NSVGgradientData* nsvg__findGradientData(NSVGparser* p, const char* id)
 	return NULL;
 }
 
-static NSVGgradient* nsvg__createGradient(NSVGparser* p, const char* id, const float* localBounds, char* paintType)
+static NSVGgradientLink* nsvg__createGradientLink(const char* id, const float *xform)
+{
+	NSVGgradientLink* grad = (NSVGgradientLink*)malloc(sizeof(NSVGgradientLink));
+	if (grad == NULL) return NULL;
```

Comment:
We should not return NULL because nsvg__createGradient(...NULL...) will crash

Reply:
You're right, that logic was flawed. Modified the checks for `NULL` in `nsvg__addShape` (where `nsvg__createGradientLink` is called) to reset `type` to `NSVG_PAINT_NONE` in this case. That way, `nsvg__createGradient` will not get called (`nsvg__assignGradients` checks for `type == NSVG_PAINT_GRADIENT_LINK`).